### PR TITLE
Task 3: Components part 1

### PR DESCRIPTION
**Pull request by Olga2157**
Wrote components implementing HTML markup for required design for home page of
InVision prototype https://projects.invisionapp.com/share/F9VXQ7IMZGY#/screens/407583174 (Only UI part).

Used <ErrorBoundary> component for catching and displaying errors.

### CODE REVIEW

File: src/components/HomePage/HomePage.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import FindSection from './FindSection';
+import ResultsFilters from './ResultsFilters';
+import MoviesCounterComponent from './MoviesCounterComponent';
+import Footer from '../shared/Footer';
+import MovieList from './MovieList';
+import AddMoviePopup from './AddMoviePopup';
+import PaginationComponent from './PaginationComponent';
+import SortByButton from './SortByButton';
+
+const HomePage: React.FC<{}> = () => {
```
Comment (Djidi1) [inline src/components/HomePage/HomePage.tsx]: А надо ли? <{}>

File: src/components/HomePage/Movie.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import BurgerMenu from './BurgerMenu';
+
+type MovieProps = {
+  movieName: string;
+  movieGenre: string;
+  movieYear: number;
+  moviePath: string;
+};
+type MovieState = {
```
Comment (Djidi1) [inline src/components/HomePage/Movie.tsx:10]: Это на будущее заготовка?

File: src/components/HomePage/PaginationComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  render() {
+    // const { someNumbersList, currentPage } = this.state;
+    return (
+
+      <Pagination id="nav-pagination">
+        <PaginationItem>
+          <PaginationLink className="pagination-link-style-arrow" previous />
+        </PaginationItem>
+        {/* Now the is just mock data in UI. Will be changed in the next tasks */}
+        <PaginationItem
```
Comment (Djidi1) [inline src/components/HomePage/PaginationComponent.tsx]: Let's move PaginationItem with PaginationLink to separate FC and use something like this:
`pages.map((item) => <Pagination item={item}>)`

File: src/components/HomePage/SortByButton.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    return (
+      <div className="row-flex">
+        <ButtonDropdown id="sort-filter"
+          toggle={function sortFunction() { }}
+        >
+          <DropdownToggle caret>
+            Sort
+          </DropdownToggle>
+          <DropdownMenu>
+            <DropdownItem divider>
```
Comment (Djidi1) [inline src/components/HomePage/SortByButton.tsx]: Items to `map`

File: src/components/shared/Heading.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+
+type HeadingProps = {
+  // headingClass?: string
+  headingText: string
+}
+type HeadingState = {
+}
+class Heading extends React.Component<HeadingProps, HeadingState> {
+  private static readonly textStyle = 'text-uppercase';
```
Comment (Djidi1) [inline src/components/shared/Heading.tsx]: weird approach for styling. Let's use separate SCSS file

File: src/components/shared/Logo.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+
+const Logo: React.FC<{}> = () => {
+  const logoClass = 'text-right  text-lowercase logo-message';
```
Comment (Djidi1) [inline src/components/shared/Logo.tsx]: if we use bootstrap styles, we can use them directly without additional variables.

File: src/styles.css [hunk 7]
```diff
@@ -0,0 +1,214 @@
+* {
```
Comment (Djidi1) [inline src/styles.css:1]: let's keep here global styles, and extract all components related styles

File: src/components/HomePage/Movie.tsx (hunk 2 above)
Comment (Olga2157) [reply src/components/HomePage/Movie.tsx:10]: да

File: src/components/HomePage/PaginationComponent.tsx (hunk 3 above)
Comment (Olga2157) [reply src/components/HomePage/PaginationComponent.tsx]: Done

File: src/components/HomePage/SortByButton.tsx (hunk 4 above)
Comment (Olga2157) [reply src/components/HomePage/SortByButton.tsx]: Fixed

File: src/components/HomePage/HomePage.tsx (hunk 1 above)
Comment (Olga2157) [reply src/components/HomePage/HomePage.tsx]: нет, по итогам митинга изменения будут отражены в следующем mr

File: src/components/shared/Heading.tsx (hunk 5 above)
Comment (Olga2157) [reply src/components/shared/Heading.tsx]: it will be fixed in the next mr

File: src/components/shared/Logo.tsx (hunk 6 above)
Comment (Olga2157) [reply src/components/shared/Logo.tsx]: ok, thanks

File: src/styles.css (hunk 7 above)
Comment (Olga2157) [reply src/styles.css:1]: as we discussed during the meeting it will be fixed in the next MR

**Pull request merged**